### PR TITLE
Model builder: Add different exit codes for different exceptions

### LINKED ISSUE #663: Return different exit codes for different errors?
Today we return exitcode 1 on all errors, e.g. on `FileNotFoundError`. We could return different error codes for different errors. These are displayed when you do `argo get ...`, so then you can quickly see what went wrong (when you know the codes), and we could maybe do something programatically for them in the future.

### PR DESCRIPTION
**Pull request by epa095**
For now we only specify the following mapping, every other exception is
mapped to 1

PermissionError: 20,
FileNotFoundError: 30

This closes #663

### CODE REVIEW

File: gordo_components/cli/cli.py [hunk 1]
```diff
@@ -181,62 +189,71 @@ def build(
...
+                )
+                saved_metadata = load_metadata(model_location)
+                all_scores = get_all_score_strings(saved_metadata)
+
+            for score in all_scores:
+                print(score)
+    except Exception as e:
+        exit_code = EXCEPTION_TO_EXITCODE.get(e.__class__, 1)
+        traceback.print_exc()
+        sys.exit(exit_code)
```
Comment (milesgranger) [inline gordo_components/cli/cli.py:256]: Small thing: add an `else` ?

File: tests/gordo_components/cli/test_cli.py [hunk 2]
```diff
@@ -224,6 +228,36 @@ def test_expand_model_complains_on_missing_vars(self):
...
+            MODEL_NAME="model-name",
+            OUTPUT_DIR=tmp_dir,
+            DATA_CONFIG=DATA_CONFIG,
+            MODEL_CONFIG=json.dumps(MODEL_CONFIG_WITH_PREDICT),
+        ):
+            result = runner.invoke(
+                cli.gordo,
+                ["build", '--evaluation-config={"cv_mode": "cross_val_only"}'],
+            )
+            assert result.exit_code == exit_code
```
Comment (milesgranger) [inline tests/gordo_components/cli/test_cli.py:258]: Tasty.

**Review by milesgranger — APPROVED** [review]
At some point, maybe we should re-evaluate this CLI's build function.. but otherwise looks acceptable to me.

File: gordo_components/cli/cli.py (hunk 1 above)
Comment (epa095) [reply gordo_components/cli/cli.py:256]: Containing what? The return?

Comment (milesgranger) [reply gordo_components/cli/cli.py:256]: yespi. Just to emphasize that two possible outcomes can happen in in this `try` block. 
I guess this is a style choice if you prefer implicit or explicit forms similar to:
```python
if foo:
    return foo
return bar
```
or 
```python
if foo:
    return foo
else:
    return bar
```

Comment (milesgranger) [reply gordo_components/cli/cli.py:256]: It's a free country, though.. :norway:

**Pull request merged**